python/ray/experimental/dag/input_node.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Dict, List
+
+from ray.experimental.dag import DAGNode
+
+
+class InputNode(DAGNode):
```

Comment:
I think we can go ahead and implement the full version, the code is something like this:

```
class InputNode:
    def __init__(self, fetch):
        self.fetch = fetch

    def _execute_impl(self, *args, **kwargs):
        if self.fetch == "__args__":
            return args
        elif self.fetch == "__kwargs__":
            return kwargs
        elif isinstance(self.fetch, int):
            return args[self.fetch]
        else:
            return kwargs[self.fetch]


class InputNodeFactory:
    args = InputNode("__args__")
    kwargs = InputNode("__kwargs__")

    def __getitem__(self, fetch):
        return InputNode(fetch)


## Use like:
# dag.input.args
# dag.input.kwargs
# dag.input[0]
# dag.input["foo"]
input = InputNodeFactory()
```

Reply:
hmm is this kinda making InputNodeFactory a singleton that produces args / kwargs accessor, how about we leave it to the next PR to properly set it up with more tests ? So for today I can get more serve pipeline side of work done for the demo.